Remove usage of BRAVE_CHROMIUM_BUILD and brave_chromium_build

mariospr opened the pull request:
It will assumed from now on that all of Brave's own code plus patches
to upstream Chromium will always be built and run as part of a Brave
build, so we can drop these guards and reduce the patching surface.

Note that github.com/brave/brave-browser/wiki/Patching-Chromium should
also be updated to reflect the change implemented in this patch.

Resolves https://github.com/brave/brave-browser/issues/12740

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

N/A

iefremov commented on patches/components-update_client-updater_state_win.cc.patch:
```diff
@@ -1,35 +1,32 @@
  
  // Google Update registry settings.
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +const wchar_t kRegPathGoogleUpdate[] = L"Software\\BraveSoftware\\Update";
```
for future work: all of these potentitally can be redefined via chromium_src?

mariospr replied:
Yes, I had the same thought exactly, but for this patch I think it's probably better to keep it like this so that it's more clear what the change is about?

mariospr replied:
@iefremov I've been looking into this today in the context of other similar situations where I could actually remove the definitions from the .patch file and put them into an override (i.e. https://github.com/brave/brave-core/pull/8063, https://github.com/brave/brave-core/pull/8061 and https://github.com/brave/brave-core/pull/8062), but in this particular case I'm not sure it's a good idea.

My reasoning is that in this case using a chromium_src override to define Brave-specific values for these variables won't work as expected, since we'd need first to redefine the ones from upstream into something like `<variable_name>_ChromiumImpl` or `<variable_name>_Unused` (depending on whether we will actually use them -e.g. for the `!defined(BRAVE_CHROMIUM_BUILD)` case- or not), but doing that will also change the places where such variables are already being referenced inside this `updater_state_win.cc` file.

If this file didn't contain any logic referencing those variables and only contained definitions (as it's the case for the PRs linked above), I'd say this approach would work. But considering that's not the case and they are indeed being referenced, I'm not sure how to do this in a way that is actually better / less bad than using a .patch file.

WDYT? Do you have any suggestion?

iefremov replied:
no suggestions, let's leave them alone :)

iefremov commented on patches/chrome-common-channel_info_win.cc.patch:
```diff
@@ -1,13 +1,13 @@
 @@ -13,7 +13,7 @@
  namespace chrome {
  
  std::string GetChannelName() {
```
candidate for chromium_src?

mariospr replied:
Indeed, will work on this (and other cases) in follow-up PRs if that's ok

mariospr replied:
Tracked in https://github.com/brave/brave-core/pull/7996

iefremov commented on patches/chrome-browser-search_engines-template_url_service_factory.cc.patch:
```diff
@@ -1,17 +1,13 @@
  
  content::BrowserContext* TemplateURLServiceFactory::GetBrowserContextToUse(
      content::BrowserContext* context) const {
-+#if defined(BRAVE_CHROMIUM_BUILD)
```
chromium_src

mariospr replied:
Ditto (re: doing this in a follow-up PR)

mariospr replied:
Tracked in https://github.com/brave/brave-core/pull/7980

iefremov commented on patches/chrome-browser-net-net_error_tab_helper.cc.patch:
```diff
@@ -1,14 +1,12 @@
  }
  
  bool NetErrorTabHelper::ProbesAllowed() const {
-+#if defined(BRAVE_CHROMIUM_BUILD)
```
maybe we could develop a way to avoid creation of certain tabhelpers, because it seems the whole thing is not needed for us

mariospr replied:
I think I lack context to properly understand what you mean here. Could you please elaborate a bit more?

iefremov replied:
It appears to me that the whole point of this class is sending some sort of "probes" which we disable in this method. So instead of patching we could maybe avoid instantiating the helper (it's unclear to me how exactly, just a raw idea). And probably there are more unneeded helpers (or keyed services) like this.

mariospr replied:
I see what you mean now, thanks for the clarification.

mariospr replied:
Handled in https://github.com/brave/brave-browser/issues/14219

iefremov reviewed (approved):
Looks good to me, and it seems that there are few patches that can be converted to chormium_src files, especially the ones with constants (I've highlighted a few, but obviously there are more)

mariospr commented:
> Looks good to me, and it seems that there are few patches that can be converted to chormium_src files, especially the ones with constants (I've highlighted a few, but obviously there are more)

@iefremov Thanks for the review! I understand you're ok with doing the chromium_src suggested changes in a separate PR, so I'll work on that next, but let me know whether there's something else you want me to change in this particular one. Thanks!

iefremov commented:
@mariospr I'm fine with this one, but it is patch masters who decide :)

mkarolin commented on chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc:
```diff
@@ -3,12 +3,6 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#if defined(BRAVE_CHROMIUM_BUILD)
 #define GOOGLE_CHROME_BUILD
```
Unrelated to this PR, but we should probably include all headers from the original file here.

mariospr replied:
I'll prepare a separate PR for these changes. Thanks

mariospr replied:
On a second thought, I think the change is small enough we can include it here (I thought there would be many other cases defining `GOOGLE_CHROME_BUILD` in other chromium_src overrides, but turns out it's only these two files). Let me know if you prefer it on a separate PR, though, but otherwise I'm merging that into this PR.

mkarolin commented on chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h:
```diff
@@ -6,14 +6,8 @@
 #define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_WEBUI_SETTINGS_METRICS_REPORTING_HANDLER_H_
 
-#if defined(BRAVE_CHROMIUM_BUILD)
 #define GOOGLE_CHROME_BUILD
```
Same, unrelated to this PR, but we should probably include all headers from the original file here.

mariospr replied:
Ditto (i.e. will do on a separate PR)

mariospr replied:
As clarified above, the change is small enough (and only affects these files) that I'm including it in this PR in the end.

mkarolin commented on patches/chrome-BUILD.gn.patch:
```diff
@@ -1,19 +1,17 @@
-+        sources -= [ "app/chrome_exe.rc", ]
-+        sources += [ "//brave/app/brave_exe.rc", ]
-+      }
++      sources -= [ "app/chrome_exe.rc", ]
++      sources += [ "//brave/app/brave_exe.rc", ]
```
Can collapse into a single line.

mariospr replied:
Hmmm... how do we do that? Note that one it's an addition and the other one is a removal :)

mariospr replied:
Ah, ok... you mean putting the two statements in the same line, not merging the two lists. Ok

mkarolin commented on patches/chrome-BUILD.gn.patch:
```diff
@@ -73,26 +71,24 @@ index 5eeb0d0c2598bbcff4981dcb6f43cae82ec37c99..d20263d25250da69f53b9b7e5322e019
-+      deps += [ "//brave:framework_bundle_data" ]
-+      if (defined(chrome_framework_version)) { framework_contents += [ "Frameworks", ] }
-+    }
++    deps += [ "//brave:framework_bundle_data" ]
++    if (defined(chrome_framework_version)) { framework_contents += [ "Frameworks", ] }
```
Can collapse into a single line.

mariospr replied:
Ok

mkarolin commented on patches/chrome-browser-profiles-profile_io_data.cc.patch:
```diff
@@ -1,14 +1,12 @@
      content::kChromeUIScheme,
      content::kChromeUIUntrustedScheme,
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +    kBraveUIScheme,
```
Unrelated to this PR, this can probably be done via a `chromium_src` override.

mariospr replied:
I actually looked into doing that yesterday, but it was not clear to me how we would do it besides creating a define in the chromium_src override and then patching this file to include that define. It would be 1 line vs 3 lines in the patch, though, so perhaps still worth it.

@mkarolin Could you clarify what you mean?

mariospr replied:
I think I better understood what you meant and got a PR already proposing a change, see https://github.com/brave/brave-core/pull/7995

mkarolin commented on patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch:
```diff
@@ -1,14 +1,13 @@
  constexpr base::FilePath::CharType kChromeExecutable[] =
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +    FILE_PATH_LITERAL("brave.exe");
-+#else
++#if 0
      FILE_PATH_LITERAL("chrome.exe");
 +#endif
```
Wonder if we can simplify this by using something like `true ? FILE_PATH_LITERAL("brave.exe") :`

mariospr replied:
That would certainly work too. I'll change it if that's preferred.

mariospr replied:
Actually, I think I'll leave it as I had it for now if you don't mind, since doing what you suggest would go against some of the rules we have in the ["Patching Chromium" wiki page](https://github.com/brave/brave-browser/wiki/Patching-Chromium):

  _"If possible write the patch to add a new line vs appending/prepending to an existing line."_

I'm aware though that your suggestion means less lines added in the patch, but my hesitation comes from the fact that, if we're going to modify the existing line, we could very well then simply replace the string doing something like this:
```diff
- FILE_PATH_LITERAL("chrome.exe");
+ FILE_PATH_LITERAL("brave.exe");
```
WDYT?

mariospr replied:
@mkarolin It doesn't seem this is blocking the merge, so I'm going to merge this PR today if it's ok to prevent it from getting conflicts (had to resolve some today already) if it's ok.

mkarolin reviewed (approved):
++
Need to update https://github.com/brave/brave-browser/wiki/Patching-Chromium re: using BRAVE_CHROMIUM_BUILD.

mariospr commented:
The error in the post-init bot (re: npm run audit_deps) is unrelated to this change and this PR can be safely merged as discussed with @mihaiplesa in the morning.

mariospr commented:
Merged and updated the wiki on https://github.com/brave/brave-browser/wiki/Patching-Chromium  (i.e. removed paragraph referencing `BRAVE_CHROMIUM_BUILD`)
